src/middlewared/middlewared/event.py

```diff
@@ -17,10 +17,14 @@ def send_event(self, etype, **kwargs):
 
     def process(self):
         self.run()
+        self.on_finish()
```

Comment:
Can we protect us from unhandled exception in `run` here?

Reply:
Good catch